packages/git/src/browser/diff/git-diff-widget.tsx

```diff
@@ -72,7 +72,9 @@ export class GitDiffWidget extends GitNavigableListWidget<GitFileChangeNode> imp
 
     async setContent(options: Git.Options.Diff) {
         this.options = options;
-        const repository = this.repositoryProvider.selectedRepository;
+        const repository = options.uri ?
+            this.repositoryProvider.findRepository(new URI(options.uri)) :
```

Comment:
I would declare a method on the repository provider. There are a lot of checks for the `option` and `option.uri`:
```js
findRepositoryOrSelected(arg: URI | string | { uri?: string | URI } | undefined): Repository | undefined {
}
```
The API is is just an idea.